components/src/dynamo/common/multimodal/image_batch_loader.py

```
@@ -0,0 +1,106 @@
+            image_futures.append(image_loader.load_image(url))
+            logger.debug(f"Preparing to load image from URL: {url[:80]}...")
+        elif isinstance(item, dict) and DECODED_VARIANT_KEY in item:
+            if enable_frontend_decoding:
+                if nixl_connector is None:
+                    logger.error(
+                        "Frontend decoding enabled but nixl_connector not provided. "
+                        "Caller must pass an initialized NIXL connector."
+                    )
+                    raise ValueError(
+                        "nixl_connector required when enable_frontend_decoding=True"
+                    )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)